packages/jellyfish-api-core/__tests__/category/loan/getCollateralToken.test.ts

```diff
@@ -0,0 +1,273 @@
+import { LoanMasterNodeRegTestContainer } from './loan_container'
+import BigNumber from 'bignumber.js'
+import { Testing } from '@defichain/jellyfish-testing'
+
+describe('Loan getCollateralToken with parameter token and height', () => {
+  const container = new LoanMasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+
+  let priceFeedId: string
+  let collateralTokenId: string
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+
+    await testing.token.create({ symbol: 'AAPL' })
+    await testing.generate(1)
+
+    priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'AAPL',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    collateralTokenId = await testing.container.call('setcollateraltoken', [{
+      token: 'AAPL',
+      factor: new BigNumber(0.5),
+      priceFeedId,
+      activateAfterBlock: 120
+    }])
+    await testing.generate(1)
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  it('should getCollateralToken', async () => {
+    const data = await testing.rpc.loan.getCollateralToken({
+      token: 'AAPL',
+      height: 120
+    })
+    expect(data).toStrictEqual({
+      [collateralTokenId]: {
+        token: 'AAPL',
+        factor: new BigNumber(0.5),
+        priceFeedId,
+        activateAfterBlock: new BigNumber(120)
+      }
+    })
+  })
+
+  it('should not getCollateralToken if token does not exist', async () => {
+    const promise = testing.rpc.loan.getCollateralToken({ token: 'TSLA', height: 120 })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Token  does not exist!\', code: -8, method: getcollateraltoken')
+  })
+
+  it('should getCollateralToken with empty string if the height is below the activation block height', async () => {
+    const data = await testing.rpc.loan.getCollateralToken({ token: 'AAPL', height: 50 })
+    expect(data).toStrictEqual({})
+  })
+
+  it('should getCollateralToken if the height is after the activation block height', async () => {
+    const data = await testing.rpc.loan.getCollateralToken({ token: 'AAPL', height: 150 })
+    expect(data).toStrictEqual({
+      [collateralTokenId]: {
+        token: 'AAPL',
+        factor: new BigNumber(0.5),
+        priceFeedId,
+        activateAfterBlock: new BigNumber(120)
+      }
+    })
+  })
+})
+
+describe('Loan getCollateralToken with parameter token only', () => {
```

Comment:
kindly refactor the test
```
describe(getCollateralToken)
  it(should getCollateralToken)
  it(should getCollateralToken with token only)
  it(should ... with height only)
  // error test
```

Reply:
They need to be in different describe to reset the block

Reply:
nope

Reply:
I think you assume activateAfterBlock to be larger if the next `it`

Reply:
more behaviour to document
create 3 collateral token
1. TSLA, 120
2. AAPL, 140
3. AMZN, 160
get(TSLA) -> return TSLA
get(140) -> return [AAPL, AMZN]
get(AMZN, 120) -> return AMZN

Reply:
@jingyi2811 i think what @canonbrother want to say is, you can setup something like
```
TSLA, 120
AAPL, 140
AMZN, 160
```
inside `beforeAll()` and then write all the test cases for `it(should getCollateralToken), it(should getCollateralToken with token only), it(should ... with height only)` within the same container. I think it's doable. Just make sure tests are independent. 
But you still have to keep `Loan getCollateralToken with no parameter` in a separate container since it needs block height adjustment and it's clearer that way alone.

Reply:
Okay refactor the code again to use 2 describes. I draw a conclusion that the number the describes you have depends on how many times you wrote `await testing.container.waitForBlockHeight(120)` [Not too sure how true is this formula]

one APPL is good enough, no need the other 2

This is how my new code looks like. 

```
describe('Loan getCollateralToken', () => {
describe('Loan getCollateralToken with parameters token or height', () => {
   describe('Loan getCollateralToken with parameter token and height', () => {
   describe('Loan getCollateralToken with parameter height only', () => {
   describe('Loan getCollateralToken with parameter token only', () => {
}   
```

Reply:
I suppose to be able to think that another 2 describes. Ya keep everything in 1 describe is impossible but not 4 describes.

Reply:
there still got duplicate tests
```
// eg: token not exists
get({token: nonexistence })
// is same as
get({ token: nonexistence: height: 200})
```

Reply:
Is there any other duplicate tests too? I want to kill 2 birds with 1 stone (Meaning Fix 2 things in 1 attempt) :)

Reply:
Fixed